### PR TITLE
Menu spec

### PR DESCRIPTION
**Pull request by rurikoaraki**
### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [ ] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes

Adding SPEC.md file for Menu explaining its various components and expected functionality

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [x] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: packages/experimental/Menu/SPEC.md [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+  pressed?: MenuItemTokens;
+}
+```
+
+#### MenuItem Slots
+
+- `root` - The outer container representing the `MenuItem` itself that wraps everything passed via the `children` prop.
+- `content` - If specified, renders the `content` prop as text.
+- `checkmark` - If specified, renders space such that the `content` aligns with other items which have checkmarks. Only used when the `Menu` has `hasCheckmarks`.
+- `submenuIndicator` - If specified, renders an SVG which indicates that the `MenuItem` is a trigger for a nested `Menu`.
```
Comment (Saadnajmi) [inline packages/experimental/Menu/SPEC.md:244]: Just noticed, there's no icon slot?

File: packages/experimental/Menu/SPEC.md [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+- Clicking on a menu's trigger will open the menu
+- Clicking outside of a menu's trigger and popover will light dismiss the menu
+- Clicking on a `MenuItem` which is not configured to track selection will invoke the `MenuItem` and close the menu by default
+- Clicking on a `MenuItem` variant which is configured to track selection will toggle its selection state in the configured way and not close the menu.
+- If the menu is configured to open on hover, it will only close once the mouse leaves both the trigger and the popover. There is a delay for this behavior.
+- Submenus are configured to open on hover by default. If the mouse stops hovering over the submenu, only the submenu will close.
+
+### Keyboard interactions
+
+- When Context menu is invoked via keyboard, open the menu and place focus on the first focusable element by default.
```
Comment (Saadnajmi) [inline packages/experimental/Menu/SPEC.md:310]: Context menu has a slightly different meaning (I think there's a context menu key on keyboards?). Did you just mean menu? I don't think this really serves like a context menu

File: packages/experimental/Menu/SPEC.md [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+- Up and down inside of the menu that has focus should navigate vertically between keyboard focusable elements
+- Up and down should circularly navigate through a list of menu items by default
```
Comment (Saadnajmi) [inline packages/experimental/Menu/SPEC.md:312]: Circular navigation on win32, but not macOS.

File: packages/experimental/Menu/SPEC.md [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+- Right key on a menu item that has an available submenu or split button should invoke the submenu, and move focus into it onto the first focusable item
+- Left key when in a submenu should close the submenu and place focus back on the parent menu's item that opened the submenu.
```
Comment (Saadnajmi) [inline packages/experimental/Menu/SPEC.md:314]: Small note that these flip in RTL?

File: packages/experimental/Menu/SPEC.md [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+- ESC on a menu should close the currently focused menu. If focus is in a submenu and ESC is hit, it should only close the current submenu and return focus to the parent's element where focus was previously.
```
Comment (Saadnajmi) [inline packages/experimental/Menu/SPEC.md:315]: On macOS, I think it's _supposed_ to close all but might in practice only close the submenu. That was a behavior difference I was ok with leaving in.

File: packages/experimental/Menu/SPEC.md [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+The default positioning for the root `Menu` is below the trigger and aligned with the left (or right in RTL) edge.
+
+#### Submenu positioning
+
+The default positioning for a submenu is to the right of the menu item trigger (or left in RTL) and aligned with the top edge.
+
+## Accessibiltiy
+
+The `MenuTrigger` will set `expand` or `collapsed` state and will respond to actions which explicitly tell it to expand or collapse (win32 only)
+The `MenuPopover` has its role set to `menu`, and `MenuItmes` and variants have their role set to `menuitem`.
```
Comment (Saadnajmi) [inline packages/experimental/Menu/SPEC.md:344]: Typo

File: packages/experimental/Menu/SPEC.md [hunk 7]
```diff
@@ -0,0 +1,345 @@
...
+
+When an option has a configurable selection state, it does not dismiss the `Menu` when toggled, as opposed to other options which close the `Menu` on being invoked by default.
+
+### Sections
+
+A `Menu` can be divided into sections using dividers. The `Menu` does not support `MenuGroups` yet, but can visually represent sections using `MenuDividers`.
+
+### Disabled option(s)
+
+All options in a `Menu` component can be disabled and should provide a visible indication for this purpose. Although disabled actions cannot be invoked, they can are keyboard focusable for the purposes of accessibility.
```
Comment (ValentinaKozlova) [inline packages/experimental/Menu/SPEC.md:104]: "they can are keyboard focusable" - is `can are` part expected?

File: packages/experimental/Menu/SPEC.md (hunk 1 above)
Comment (rurikoaraki) [reply packages/experimental/Menu/SPEC.md:244]: Yeah, icon's out of scope for V1.

File: packages/experimental/Menu/SPEC.md (hunk 7 above)
Comment (rurikoaraki) [reply packages/experimental/Menu/SPEC.md:104]: Ah thanks for the catch

File: packages/experimental/Menu/SPEC.md (hunk 6 above)
Comment (rurikoaraki) [reply packages/experimental/Menu/SPEC.md:344]: Thanks for catching

**Pull request merged**